src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -226,266 +226,189 @@ public void initForSectionPanelViews(InstructorAttributes instructor,
-                    } else {
-                        buildMissingParticipantPanelsForSection(sectionPanel, prevSection, participantsWithResponses);
-                        participantsWithResponses.clear();
-                    }
-                    
-                    // add to sectionPanels,
-                    sectionPanels.put(prevSection, sectionPanel);
-                    sectionsWithResponses.add(prevSection);
-                    
-                    // setup for next section
-                    teamsWithResponses.clear();
-                    teamsWithResponses.add(currentTeam);
-                    
-                    sectionPanel = new InstructorFeedbackResultsSectionPanel();
-                }
-            }
-            
-            if (isDifferentTeam) {
                 sectionPanel.getIsTeamWithResponses().put(currentTeam, true);
             }
-            
+
             // Build participant panel for the current primary participant
-            InstructorFeedbackResultsParticipantPanel primaryParticipantPanel =
-                    buildGroupByParticipantPanel(primaryParticipantIdentifier,
-                                                 primaryToSecondaryParticipantToResponsesMap,
-                                                 additionalInfoId, primaryParticipantIndex);
-            
-            if (isGroupedByTeam()) {
-                sectionPanel.addParticipantPanel(currentTeam, primaryParticipantPanel);
-            } else {
-                sectionPanel.addParticipantPanel(primaryParticipantPanel);
-            }
-            
-            participantsWithResponses.add(primaryParticipantIdentifier);
-            
+            InstructorFeedbackResultsParticipantPanel recipientPanel =
+                    buildGroupByParticipantPanel(primaryParticipantIdentifier, primaryToSecondaryParticipantToResponsesMap,
+                                                 sectionPrefix + additionalInfoId, primaryParticipantIndex);
+
+            sectionPanel.addParticipantPanel(currentTeam, recipientPanel);
+
+            teamMembersWithResponses.add(primaryParticipantIdentifier);
+
             prevTeam = currentTeam;
-            prevSection = currentSection;
         }
-        
+
         // for the last section with responses
-        if (isGroupedByTeam()) {
-            buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, participantsWithResponses);
-        }
-        
+        buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, teamMembersWithResponses);
+
         teamsWithResponses.add(prevTeam);
-        if (isGroupedByTeam()) {
-            buildMissingTeamAndParticipantPanelsForSection(
-                                            sectionPanel, prevSection, teamsWithResponses);
-        } else {
-            buildMissingParticipantPanelsForSection(sectionPanel, prevSection, participantsWithResponses);
```

Comment:
When not grouped by teams, the missing participants are incorrectly generated. This can be reproduced when the students are not ordered in the teams alphabetically.